### ISSUE TITLE
[FE] Logged-in homepage:  Investigate level of effort for showing GI Bill benefits balance

### ISSUE THREAD

**Issue by Samara-Strauss** [department-of-veterans-affairs/va.gov-team#10255]
## Background

GI bill benefit balances information is quite popular — we can see in analytics that this information is the 9th most visited page on the modernized VA.gov for logged-in users, not to mention user research participants mentioned this was something they wanted to see on their logged-in homepage if that benefit applied to them.

However, despite its popularity, we do not show this info on My VA. We link to the tool, but we do not show the balance outright. We should investigate why this is. I think there are some reliability issues with the tool, but we should confirm that and determine whether this can be included on the logged-in homepage as part of V1 or a later phase.

## Tasks

- [ ] Investigate how the GI bill benefits balance works.
- [ ] Determine the level of effort for showing this on the logged-in homepage.
- [ ] Make a recommendation as to whether we can include this in V1 or a later phase of this work.

## Acceptance criteria

- [ ] We have determined whether showing GI Bill benefit balances on the logged-in homepage will be part of V1 of the logged-in homepage.

**Comment by erikphansen** [comment]
- currently this data is shown at https://staging.va.gov/education/gi-bill/post-9-11/ch-33-benefit/status
- it uses the `GET /v0/post911_gi_bill_status`

- [ ] find a staging user that has data; user 36 failed/the React app crashed
- [ ] look at the source code for the React app that lives at https://staging.va.gov/education/gi-bill/post-9-11/ch-33-benefit/status

**Comment by erikphansen** [comment]
[This doc](https://github.com/department-of-veterans-affairs/va.gov-team/tree/master/products/education-careers/statement-of-benefits) says user 229 should work at `https://staging.va.gov/education/gi-bill/post-9-11/ch-33-benefit/status` but that user gets nothing from the `GET /v0/post911_gi_bill_status` endpoint:

```json
{
  "data": {
    "id": "",
    "type": "evss_gi_bill_status_gi_bill_status_responses",
    "attributes": {
      "firstName": null,
      "lastName": null,
      "nameSuffix": null,
      "dateOfBirth": null,
      "vaFileNumber": null,
      "regionalProcessingOffice": null,
      "eligibilityDate": null,
      "delimitingDate": null,
      "percentageBenefit": null,
      "originalEntitlement": null,
      "usedEntitlement": null,
      "remainingEntitlement": null,
      "activeDuty": null,
      "veteranIsEligible": null,
      "enrollments": null
    }
  },
  "meta": { "status": "SERVER_ERROR" }
}
```

Will need to follow up again with a user that works.

**Comment by erikphansen** [comment]
FWIW, based on [the Swagger docs](https://department-of-veterans-affairs.github.io/va-digital-services-platform-docs/api-reference/#/benefits_status/getPost911GiBillStatus) it looks like the benefit data is described in terms of days and months remaining. I'm not sure what to make of that.

```json
{
  "meta": {
    "status": "OK"
  },
  "data": {
    "id": "string",
    "type": "evss_gi_bill_status_gi_bill_status_responses",
    "attributes": {
      "first_name": "Abraham",
      "last_name": "Lincoln",
      "name_suffix": "Jr",
      "date_of_birth": "1955-11-12T06:00:00.000+0000",
      "va_file_number": "123456789",
      "regional_processing_office": "Central Office Washington, DC",
      "eligibility_date": "2004-10-01T04:00:00.000+0000",
      "delimiting_date": "2015-10-01T04:00:00.000+0000",
      "percentage_benefit": 100,
      "veteran_is_eligible": false,
      "active_duty": false,
      "original_entitlement": {
        "days": 0,
        "months": 0
      },
      "used_entitlement": {
        "days": 0,
        "months": 0
      },
      "remaining_entitlement": {
        "days": 0,
        "months": 0
      },
      "enrollments": [
        {
          "begin_date": "2012-11-01T04:00:00.000+00:00",
          "end_date": "2012-12-01T05:00:00.000+00:00",
          "facility_code": "12345678",
          "facility_name": "Purdue University",
          "participant_id": "11170323",
          "training_type": "UNDER_GRAD",
          "term_id": {},
          "hour_type": {},
          "full_time_hours": 12,
          "full_time_credit_hour_under_grad": {},
          "vacation_day_count": 0,
          "on_campus_hours": 12,
          "online_hours": 0,
          "yellow_ribbon_amount": 0,
          "status": "Approved",
          "amendments": [
            {
              "on_campus_hours": "Unknown Type: number,null",
              "online_hours": "Unknown Type: number,null",
              "yellow_ribbon_amount": "Unknown Type: number,null",
              "type": "string",
              "change_effective_date": "Unknown Type: string,null"
            }
          ]
        }
      ]
    }
  }
}
```

**Comment by Samara-Strauss** [comment]
My guess is that this will be too complicated to handle in a V1. What are your thoughts? How much more investigation needs to happen here?

**Comment by erikphansen** [comment]
I think that this should be doable. But it's concerning that I can't find a staging user that can view the page. Anna said [this list of users](https://github.com/department-of-veterans-affairs/va.gov-team-sensitive/blob/master/Administrative/vagov-users/mvi-staging-users-birls-participant-edipi.csv) should work, but they all get the same error I posted above. More concerning is that the page crashes and shows nothing when the network gives that response. That's a bug that needs to get fixed by someone but I don't know who is responsible for that app right now.

In summary:
- `GET https://staging-api.va.gov/v0/post911_gi_bill_status` is always returning an error
- The React app does not properly handle that error and crashes/shows nothing

**Comment by Samara-Strauss** [comment]
I can ask about this in Slack and see if anyone on BAM2 is aware of a staging user that works.

**Comment by erikphansen** [comment]
Based on this thread and yesterday's standup it sounds like this one will be closed for now.
